packages/extensions/src/placeholder/index.ts

```diff
@@ -1,13 +1,13 @@
 import { definePlugin, isInCodeBlock } from '@prosekit/core'
-import type { ProseMirrorNode } from '@prosekit/pm/model'
+import { type ProseMirrorNode, type Node } from '@prosekit/pm/model'
 import { type EditorState, Plugin, PluginKey } from '@prosekit/pm/state'
 import { Decoration, DecorationSet } from '@prosekit/pm/view'
 
 export interface PlaceholderOptions {
   /**
    * The placeholder text to use.
    */
-  placeholder: string
+  placeholder: ((node: Node) => string) | string
```

Comment:
```suggestion
  placeholder: ((options: {node: ProseMirrorNode}) => string) | string
```

Let's make the parameter an object, just in case we want to add more thing to it such as `pos` in the future.

Reply:
That's a good point. Update at https://github.com/prosekit/prosekit/pull/808/commits/c4a6d73232899a062a5e60c2c9639d835b8c771e